code/xcvm/cosmwasm/contracts/order/src/lib.rs

```diff
@@ -1,27 +1,26 @@
-#![feature(result_flattening)]
-mod error;
+#![allow(clippy::disallowed_methods)] // does unwrap inside
 
 use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{
-	wasm_execute, Addr, BankMsg, BlockInfo, Coin, Order, StdError, Uint128, Uint64,
+	wasm_execute, Addr, BankMsg, Coin, Event, Order, StdError, Storage, Uint128, Uint64,
 };
 use cw_storage_plus::{Index, IndexList, IndexedMap, Item, Map, MultiIndex};
-use itertools::*;
 use sylvia::{
 	contract,
 	cw_std::{ensure, Response, StdResult},
 	entry_points,
 	types::{ExecCtx, InstantiateCtx, QueryCtx},
 };
-use xc_core::{service::dex::ExchangeId, shared::Displayed, NetworkId};
 
 /// so this is just to make code easy to read, we will optimize later
 use num_rational::BigRational;
 
-use crate::error::ContractError;
-
-pub type Amount = Displayed<u128>;
-pub type OrderId = Displayed<u128>;
+// fix core in std in contract
+// use xc_core::{service::dex::ExchangeId, shared::Displayed, NetworkId};
+pub type ExchangeId = Uint128;
+pub type Amount = u128;
+pub type OrderId = Uint128;
```

Comment:
I find this confusing. Why is ExchangeId set to Uint128 while Amount is u128? It seems to me that those aliases should all either be plain Rust type or Uint128 and then in structs we should accordingly either use Uint128 or those aliases.

Reply:
Sylvia does not compile messages with u128, only Uint128.

Reply:
will try with displayed after making it compile in wasm std

Reply:
Sure, the messages should use Uint128 but my point is that either consistently use Uint128 in the messages or declare those aliases to be Uint128 and use them. The way here some are Uint128 and some are u128 is confusing and misleading.